### ISSUE TITLE
All document properties are wipeout in Excel addin

### ISSUE THREAD

**Issue by jag-cloudextend** [OfficeDev/office-js#2972]
# Provide required information needed to triage your issue
We have an application that makes use of [Office Settings](https://learn.microsoft.com/en-us/javascript/api/office/office.settings?view=excel-js-1.9). Recently all these properties are wipeout and we are getting back null
## Your Environment
This is happening on all the platforms
 
## Expected behavior
I should be able to retrieve the document properties

## Current behavior
Whenever we get the property, it returns null

## Steps to reproduce

1. Open the addin 
2. Set few properties in settings
3. Reload the app and fetch the settings

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by samantharamon** [comment]
Thank you for reporting this, @Jagapathi126.

@RuizhiSunMS are you able to assist with this?

**Comment by RuizhiSunMS** [comment]
Thanks @samantharamon .
Hi @Jagapathi126 , can you share your addin with us? It may help in locating the cause.

**Comment by jag-cloudextend** [comment]
@RuizhiSunMS Thank you for your response. Can you provide your email id? I will share more details in an email

**Comment by RuizhiSunMS** [comment]
ruizhisun@microsoft.com
You can send with this address. Thanks.

**Comment by jag-cloudextend** [comment]
Thank you for the address. I have shared the details over email.

**Comment by JinghuiMS** [comment]
@Jagapathi126 
Thanks for reaching us! I'd like to confirm if you/your customer hits this setting issue on Excel web? If so, please check it again on Excel online, this issue should be gone! Feel free to ask me if you have further questions.

**Comment by jag-cloudextend** [comment]
Does that mean there was an issue in Excel web which is resolved now? Do we have a way to restore those settings?

**Comment by JinghuiMS** [comment]
yes, there was a problem with the build version in early November on Excel online, but it has been resolved now. You can refer to https://learn.microsoft.com/en-us/javascript/api/office/office.settings?view=excel-js-1.9 for how to restore the settings.

**Comment by jag-cloudextend** [comment]
Thank you @JinghuiMS for the update. However, I am unable to locate a way to restore the missed settings. Can you help me with more details?

**Comment by JinghuiMS** [comment]
To clarify that we need to confirm firstly that you can get your settings now. You should ensure you set/get setting from document by using the same add-in. If you still can't get the setting now, this issue should be the new one to us. We need to do new investigation for it.
For restore the missed settings, I'm afraid we don't support this feature. BTW, the issue happened in early November didn't affect the set setting to document.

**Comment by jag-cloudextend** [comment]
@JinghuiMS we are able to set and get the settings now. 
We are highly dependent on Settings. Do you have any recommendations to avoid this situation in future?

**Comment by JinghuiMS** [comment]
@Jagapathi126 
Sorry for the inconvenience caused to you. We can totally understand your concern! We will try our best to ensure the quality of our products. At the same time, if you encounter any problems, please report to us in time, and we will deal with them ASAP! I'd like to know what's the number of potential customers for the add-in? You can email us if you‘re not convenient to say it here! 
Thanks again for reporting your issues!

**Comment by ghost** [comment]
This issue has been closed due to inactivity. Please comment if you still need assistance and we'll re-open the issue.

**Comment by guoms1** [comment]
Hi, @jag-cloudextend 

Thank you for bringing this issue back to our attention. After investigating, I couldn’t reproduce the problem you described. I tested it on several platforms—Word Online, Word Desktop, and Excel Online—and none of them showed the issue. You can see this in the attached screen recording. Additionally, I’ve provided the ScriptLab script for you to try; just follow the steps in the video.

[GH2972-Excel.txt](https://github.com/user-attachments/files/19025189/GH2972-Excel.txt)
[GH2972-Word.txt](https://github.com/user-attachments/files/19025194/GH2972-Word.txt)

https://github.com/user-attachments/assets/81f2ac07-8469-4747-8734-87963546eef6
